source/profiles.c

```diff
@@ -122,26 +133,31 @@ void animatedWave(led_t *currentKeyLedColors, uint8_t intensity) {
                       intensity);
     waveValue[i] += waveDirection[i];
   }
+  return false;
 }
 
 uint8_t animatedPressedFadeBuf[NUM_ROW * NUM_COLUMN] = {0};
 
-void reactiveFade(led_t *ledColors, uint8_t intensity) {
+bool reactiveFade(led_t *ledColors, uint8_t intensity) {
+  bool shouldUpdate = false;
```

Comment:
The function signature changed to `bool` because of this. For "reactive" profiles if there are no LEDs to be enabled we can suspend the current thread to prevent useless looping which should improve the battery life.

Reply:
OK